### ISSUE TITLE
Question: how to force install python-version (without build C-dependency code)

### ISSUE THREAD

**Issue by BooBSD** [ClickHouse/clickhouse-connect#44]
Hello,
I have a problem during installation process under Docker and I have a question:
Is it possible to force install python-version of `clickhouse-connect` without building C-dependency code?

**Comment by genzgd** [comment]
It *should* still install correctly if the C build fails, but that hasn't been tested extensively.  Can you provide a snippet of the docker file where it fails?

**Comment by genzgd** [comment]
The other possible solution is to build a many_linux wheel, I'll try to look at that early next week.

**Comment by BooBSD** [comment]
@genzgd Thank you for your answer.

`Dockerfile`:

```
FROM python:3.8-slim-buster as slim

ENV SRC_ROOT="/code/src"
ENV BUILD_ROOT="/code/build"

RUN pip install -U pip==22.2.2
...
COPY requirements $SRC_ROOT/requirements
RUN cd $SRC_ROOT/ && pip install -r $SRC_ROOT/requirements/slim.txt  -t $BUILD_ROOT/
```

The result:

```
#18 25.10   Building wheel for clickhouse-connect (pyproject.toml): started
#18 25.51   Building wheel for clickhouse-connect (pyproject.toml): finished with status 'error'
#18 25.53   error: subprocess-exited-with-error
#18 25.53   
#18 25.53   × Building wheel for clickhouse-connect (pyproject.toml) did not run successfully.
#18 25.53   │ exit code: 1
#18 25.53   ╰─> [70 lines of output]
#18 25.53       running bdist_wheel
#18 25.53       running build
#18 25.53       running build_py
#18 25.53       creating build
#18 25.53       creating build/lib.linux-x86_64-cpython-38
#18 25.53       creating build/lib.linux-x86_64-cpython-38/clickhouse_connect
#18 25.53       copying clickhouse_connect/entry_points.py -> build/lib.linux-x86_64-cpython-38/clickhouse_connect
#18 25.53       copying clickhouse_connect/json_impl.py -> build/lib.linux-x86_64-cpython-38/clickhouse_connect
#18 25.53       copying clickhouse_connect/__init__.py -> build/lib.linux-x86_64-cpython-38/clickhouse_connect
#18 25.53       creating build/lib.linux-x86_64-cpython-38/clickhouse_connect/cc_sqlalchemy
#18 25.53       copying clickhouse_connect/cc_sqlalchemy/dialect.py -> build/lib.linux-x86_64-cpython-38/clickhouse_connect/cc_sqlalchemy
#18 25.53       copying clickhouse_connect/cc_sqlalchemy/__init__.py -> build/lib.linux-x86_64-cpython-38/clickhouse_connect/cc_sqlalchemy
#18 25.53       copying clickhouse_connect/cc_sqlalchemy/reflect.py -> build/lib.linux-x86_64-cpython-38/clickhouse_connect/cc_sqlalchemy
#18 25.53       creating build/lib.linux-x86_64-cpython-38/clickhouse_connect/driver
#18 25.53       copying clickhouse_connect/driver/extras.py -> build/lib.linux-x86_64-cpython-38/clickhouse_connect/driver
#18 25.53       copying clickhouse_connect/driver/parser.py -> build/lib.linux-x86_64-cpython-38/clickhouse_connect/driver
#18 25.53       copying clickhouse_connect/driver/models.py -> build/lib.linux-x86_64-cpython-38/clickhouse_connect/driver
#18 25.53       copying clickhouse_connect/driver/exceptions.py -> build/lib.linux-x86_64-cpython-38/clickhouse_connect/driver
#18 25.53       copying clickhouse_connect/driver/httpadapter.py -> build/lib.linux-x86_64-cpython-38/clickhouse_connect/driver
#18 25.53       copying clickhouse_connect/driver/native.py -> build/lib.linux-x86_64-cpython-38/clickhouse_connect/driver
#18 25.53       copying clickhouse_connect/driver/rowbinary.py -> build/lib.linux-x86_64-cpython-38/clickhouse_connect/driver
#18 25.53       copying clickhouse_connect/driver/common.py -> build/lib.linux-x86_64-cpython-38/clickhouse_connect/driver
#18 25.53       copying clickhouse_connect/driver/httpclient.py -> build/lib.linux-x86_64-cpython-38/clickhouse_connect/driver
#18 25.53       copying clickhouse_connect/driver/__init__.py -> build/lib.linux-x86_64-cpython-38/clickhouse_connect/driver
#18 25.53       copying clickhouse_connect/driver/options.py -> build/lib.linux-x86_64-cpython-38/clickhouse_connect/driver
#18 25.53       copying clickhouse_connect/driver/query.py -> build/lib.linux-x86_64-cpython-38/clickhouse_connect/driver
#18 25.53       copying clickhouse_connect/driver/ddl.py -> build/lib.linux-x86_64-cpython-38/clickhouse_connect/driver
#18 25.53       copying clickhouse_connect/driver/transform.py -> build/lib.linux-x86_64-cpython-38/clickhouse_connect/driver
#18 25.53       copying clickhouse_connect/driver/threads.py -> build/lib.linux-x86_64-cpython-38/clickhouse_connect/driver
#18 25.53       copying clickhouse_connect/driver/client.py -> build/lib.linux-x86_64-cpython-38/clickhouse_connect/driver
#18 25.53       creating build/lib.linux-x86_64-cpython-38/clickhouse_connect/datatypes
#18 25.53       copying clickhouse_connect/datatypes/string.py -> build/lib.linux-x86_64-cpython-38/clickhouse_connect/datatypes
#18 25.53       copying clickhouse_connect/datatypes/format.py -> build/lib.linux-x86_64-cpython-38/clickhouse_connect/datatypes
#18 25.53       copying clickhouse_connect/datatypes/special.py -> build/lib.linux-x86_64-cpython-38/clickhouse_connect/datatypes
#18 25.53       copying clickhouse_connect/datatypes/base.py -> build/lib.linux-x86_64-cpython-38/clickhouse_connect/datatypes
#18 25.53       copying clickhouse_connect/datatypes/registry.py -> build/lib.linux-x86_64-cpython-38/clickhouse_connect/datatypes
#18 25.53       copying clickhouse_connect/datatypes/numeric.py -> build/lib.linux-x86_64-cpython-38/clickhouse_connect/datatypes
#18 25.53       copying clickhouse_connect/datatypes/network.py -> build/lib.linux-x86_64-cpython-38/clickhouse_connect/datatypes
#18 25.53       copying clickhouse_connect/datatypes/__init__.py -> build/lib.linux-x86_64-cpython-38/clickhouse_connect/datatypes
#18 25.53       copying clickhouse_connect/datatypes/temporal.py -> build/lib.linux-x86_64-cpython-38/clickhouse_connect/datatypes
#18 25.53       copying clickhouse_connect/datatypes/container.py -> build/lib.linux-x86_64-cpython-38/clickhouse_connect/datatypes
#18 25.53       creating build/lib.linux-x86_64-cpython-38/clickhouse_connect/dbapi
#18 25.53       copying clickhouse_connect/dbapi/cursor.py -> build/lib.linux-x86_64-cpython-38/clickhouse_connect/dbapi
#18 25.53       copying clickhouse_connect/dbapi/__init__.py -> build/lib.linux-x86_64-cpython-38/clickhouse_connect/dbapi
#18 25.53       copying clickhouse_connect/dbapi/connection.py -> build/lib.linux-x86_64-cpython-38/clickhouse_connect/dbapi
#18 25.53       creating build/lib.linux-x86_64-cpython-38/clickhouse_connect/driverc
#18 25.53       copying clickhouse_connect/driverc/__init__.py -> build/lib.linux-x86_64-cpython-38/clickhouse_connect/driverc
#18 25.53       creating build/lib.linux-x86_64-cpython-38/clickhouse_connect/cc_superset
#18 25.53       copying clickhouse_connect/cc_superset/datatypes.py -> build/lib.linux-x86_64-cpython-38/clickhouse_connect/cc_superset
#18 25.53       copying clickhouse_connect/cc_superset/engine.py -> build/lib.linux-x86_64-cpython-38/clickhouse_connect/cc_superset
#18 25.53       copying clickhouse_connect/cc_superset/__init__.py -> build/lib.linux-x86_64-cpython-38/clickhouse_connect/cc_superset
#18 25.53       creating build/lib.linux-x86_64-cpython-38/clickhouse_connect/cc_sqlalchemy/ddl
#18 25.53       copying clickhouse_connect/cc_sqlalchemy/ddl/custom.py -> build/lib.linux-x86_64-cpython-38/clickhouse_connect/cc_sqlalchemy/ddl
#18 25.53       copying clickhouse_connect/cc_sqlalchemy/ddl/__init__.py -> build/lib.linux-x86_64-cpython-38/clickhouse_connect/cc_sqlalchemy/ddl
#18 25.53       copying clickhouse_connect/cc_sqlalchemy/ddl/tableengine.py -> build/lib.linux-x86_64-cpython-38/clickhouse_connect/cc_sqlalchemy/ddl
#18 25.53       creating build/lib.linux-x86_64-cpython-38/clickhouse_connect/cc_sqlalchemy/sql
#18 25.53       copying clickhouse_connect/cc_sqlalchemy/sql/__init__.py -> build/lib.linux-x86_64-cpython-38/clickhouse_connect/cc_sqlalchemy/sql
#18 25.53       copying clickhouse_connect/cc_sqlalchemy/sql/preparer.py -> build/lib.linux-x86_64-cpython-38/clickhouse_connect/cc_sqlalchemy/sql
#18 25.53       copying clickhouse_connect/cc_sqlalchemy/sql/ddlcompiler.py -> build/lib.linux-x86_64-cpython-38/clickhouse_connect/cc_sqlalchemy/sql
#18 25.53       creating build/lib.linux-x86_64-cpython-38/clickhouse_connect/cc_sqlalchemy/datatypes
#18 25.53       copying clickhouse_connect/cc_sqlalchemy/datatypes/sqltypes.py -> build/lib.linux-x86_64-cpython-38/clickhouse_connect/cc_sqlalchemy/datatypes
#18 25.53       copying clickhouse_connect/cc_sqlalchemy/datatypes/base.py -> build/lib.linux-x86_64-cpython-38/clickhouse_connect/cc_sqlalchemy/datatypes
#18 25.53       copying clickhouse_connect/cc_sqlalchemy/datatypes/__init__.py -> build/lib.linux-x86_64-cpython-38/clickhouse_connect/cc_sqlalchemy/datatypes
#18 25.53       running build_ext
#18 25.53       building 'creaders' extension
#18 25.53       creating build/temp.linux-x86_64-cpython-38
#18 25.53       creating build/temp.linux-x86_64-cpython-38/clickhouse_connect
#18 25.53       creating build/temp.linux-x86_64-cpython-38/clickhouse_connect/driverc
#18 25.53       gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -fPIC -I/usr/local/include/python3.8 -c clickhouse_connect/driverc/creaders.c -o build/temp.linux-x86_64-cpython-38/clickhouse_connect/driverc/creaders.o
#18 25.53       error: command 'gcc' failed: No such file or directory
#18 25.53       [end of output]
#18 25.53   
#18 25.53   note: This error originates from a subprocess, and is likely not a problem with pip.
#18 25.53   ERROR: Failed building wheel for clickhouse-connect
#18 25.53 Failed to build clickhouse-connect
#18 25.53 ERROR: Could not build wheels for clickhouse-connect, which is required to install pyproject.toml-based projects
#18 ERROR: executor failed running [/bin/sh -c cd $SRC_ROOT/ && pip install -r $SRC_ROOT/requirements/slim.txt  -t $BUILD_ROOT/]: runc did not terminate sucessfully
```

**Comment by genzgd** [comment]
Thanks . . . I'll try to get that fixed in the next few days, at a minimum it should fall back to the pure python implementation.

In the meantime if you are interested there are some workarounds such as using a docker image with the build tools or doing a direct checkout of ClickHouse Connect from Github and running setup.py instead of installing via pip.

**Comment by genzgd** [comment]
This is fixed in the new release 0.2.8. I've added binary wheels that should be installed for almost any Linux distribution, as well as fixed a bug where setup.py was not correctly falling back to pure python when build tools were not present.
